seatunnel-engine/seatunnel-engine-core/src/main/java/org/apache/seatunnel/engine/core/dag/logical/LogicalDag.java

```diff
@@ -129,40 +141,47 @@ public int getClassId() {
     @Override
     public void writeData(ObjectDataOutput out) throws IOException {
         out.writeInt(logicalVertexMap.size());
-
         for (Map.Entry<Long, LogicalVertex> entry : logicalVertexMap.entrySet()) {
             out.writeLong(entry.getKey());
             out.writeObject(entry.getValue());
         }
 
         out.writeInt(edges.size());
-
         for (LogicalEdge edge : edges) {
             out.writeObject(edge);
         }
 
+        out.writeInt(saveModeList.size());
+        for (SupportSaveMode saveMode : saveModeList) {
```

Comment:
Should we consider the DataSerializable of this?

Reply:
In fact, the current SupportSaveMode declarations are all on the SeaTunnelSink implementation class, and all SeaTunnelSinks are serializable by default. It is also engine-independent.